### PR TITLE
feat(privacy report): merge subjects and third party reports

### PR DESCRIPTION
**Pull request by elsapet**
## Description
Following product feedback and manual testing: 

* Merge subjects and third party reports into one "privacy" report
* Bug fixes for third party report section (counts were wrong; multiple subjects were not handled)
* Add "New Relic" as associated recipe to new relic rule

References #398

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/flag/report_flags.go [hunk 1]
```diff
@@ -11,12 +11,11 @@ var (
...
 	FormatEmpty = ""
 
-	ReportDetectors  = "detectors" // nodoc: internal report type
-	ReportDataFlow   = "dataflow"
-	ReportPrivacy    = "privacy"
-	ReportThirdParty = "thirdparty"
-	ReportSummary    = "summary"
-	ReportStats      = "stats"
+	ReportDetectors = "detectors" // nodoc: internal report type
+	ReportDataFlow  = "dataflow"
```
Comment (cfabianski) [inline pkg/flag/report_flags.go:15]: dataflow should be removed from the documentation (could be kept as internal for now) now that we have the privacy report :)

File: pkg/flag/report_flags.go [hunk 2]
```diff
@@ -11,12 +11,11 @@ var (
...
+	ReportPrivacy   = "privacy"
+	ReportSummary   = "summary"
+	ReportStats     = "stats"
```
Comment (cfabianski) [inline pkg/flag/report_flags.go:18]: stats too

File: pkg/flag/report_flags.go (hunk 1 above)
Comment (elsapet) [reply pkg/flag/report_flags.go:15]: @cfabianski would you be happy if we address this in a separate PR, which also adds the privacy report documentation?

Comment (cfabianski) [reply pkg/flag/report_flags.go:15]: sure

**Review by cfabianski — APPROVED** [review]

**Comment by elsapet** [comment]
Thanks @cfabianski 🙏 Draft PR for the docs changes is here https://github.com/Bearer/curio/pull/494

**Pull request merged**